tests/api.rs

```diff
@@ -226,6 +226,73 @@ impl scroll::ctx::FromCtx<scroll::Endian> for Bar {
         }
     }
 }
+#[derive(Debug)]
+struct StrDrop<'a> {
+    #[allow(unused)]
+    s: &'a str,
+}
+
+impl<'a> scroll::ctx::TryFromCtx<'a, scroll::Endian> for StrDrop<'a> {
+    type Error = scroll::Error;
+    fn try_from_ctx(
+        bytes: &'a [u8],
+        _ctx: scroll::Endian,
+    ) -> ::std::result::Result<(Self, usize), Self::Error> {
+        let mut offset = 0;
+        let s = bytes.gread(&mut offset)?;
+        Ok((Self { s }, offset))
+    }
+}
+
+impl<'a> Drop for StrDrop<'a> {
+    fn drop(&mut self) {
+        println!("Dropping {self:?}");
+    }
+}
+
+#[derive(Debug)]
+struct StringDrop {
+    #[allow(unused)]
+    s: String,
+}
+
+impl<'a> scroll::ctx::TryFromCtx<'a, scroll::Endian> for StringDrop {
+    type Error = scroll::Error;
+    fn try_from_ctx(
+        bytes: &'a [u8],
+        _ctx: scroll::Endian,
+    ) -> ::std::result::Result<(Self, usize), Self::Error> {
+        let mut offset = 0;
+        let s: &str = bytes.gread(&mut offset)?;
+        Ok((Self { s: String::from(s) }, offset))
+    }
+}
+
+impl Drop for StringDrop {
+    fn drop(&mut self) {
+        println!("Dropping {self:?}");
```

Comment:
technically i think you need to call `drop(&mut self.s)`

Reply:
I removed it, since the compiler emitted a warning stating, that dropping a reference doesn't actually do anything.

Reply:
Here is the relevant doc link: https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#dropping-references

Reply:
afaik this is a leak in our VecDrop, since you aren't dropping the owned Vec here. The above would apply to the BytesDrop, which is a reference, as you say, which doesn't need to be dropped technically. Since this is a test it doesn't really matter and can be fixed in a followup, but i'm very sure this is a memory leak in this drop impl.

Reply:
Actually, the fields of the struct are still dropped by the compiler, see: https://doc.rust-lang.org/std/ops/trait.Drop.html#examples

Reply:
ah thank you, I will give it a read, haven't written drop explicitly in a long while, so good to know :)